pkg/api/store/globaldns/globaldns_store.go

```diff
@@ -0,0 +1,86 @@
+package globaldns
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/norman/types/convert"
+	managementv3 "github.com/rancher/types/client/management/v3"
+)
+
+func Wrap(store types.Store) types.Store {
+	storeWrapped := &Store{
+		Store: store,
+	}
+	storeWrapped.mu = sync.Mutex{}
+	return storeWrapped
+}
+
+type Store struct {
+	types.Store
+	mu sync.Mutex
+}
+
+func (p *Store) Create(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}) (map[string]interface{}, error) {
+	fqdn := convert.ToString(data["fqdn"])
+	if fqdn == "" {
+		return nil, httperror.NewFieldAPIError(httperror.MissingRequired, "FQDN", "")
+	}
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	if err := canUseFQDN(apiContext, fqdn); err != nil {
+		return nil, err
+	}
+
+	return p.Store.Create(apiContext, schema, data)
+}
+
+func (p *Store) Update(apiContext *types.APIContext, schema *types.Schema, data map[string]interface{}, id string) (map[string]interface{}, error) {
+	updatedFQDN := convert.ToString(data["fqdn"])
+	if updatedFQDN == "" {
+		return nil, httperror.NewFieldAPIError(httperror.MissingRequired, "FQDN", "")
+	}
+
+	existingGlobalDNS, err := p.ByID(apiContext, schema, id)
+	if err != nil {
+		return nil, err
+	}
+
+	fqdn, ok := existingGlobalDNS["fqdn"].(string)
+	if !ok {
+		fqdn = ""
```

Comment:
when would that be the case? when fqdn is not set on a global dns entry

Reply:
yeah relaxing the check, since its a required field